protocol/communities_messenger_test.go

```diff
@@ -3684,3 +3684,239 @@ func (s *MessengerCommunitiesSuite) TestRetrieveBigCommunity() {
+	s.Require().NotNil(notification)
+	s.Require().Equal(notification.Type, ActivityCenterNotificationTypeCommunityMembershipRequest)
+	s.Require().Equal(notification.MembershipStatus, ActivityCenterMembershipStatusPending)
+	s.Require().Equal(notification.Read, false)
+	s.Require().Equal(notification.Accepted, false)
+	s.Require().Equal(notification.Dismissed, false)
+
+	// Accept request
+
+	acceptRequestToJoin := &requests.AcceptRequestToJoinCommunity{ID: requestToJoin1.ID}
+
+	response, err = s.owner.AcceptRequestToJoinCommunity(acceptRequestToJoin)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+
+	s.Require().Len(response.Communities(), 1)
+
+	updatedCommunity := response.Communities()[0]
+
+	s.Require().NotNil(updatedCommunity)
+	s.Require().True(updatedCommunity.HasMember(&s.alice.identity.PublicKey))
+
+	s.Require().Len(response.ActivityCenterNotifications(), 1)
+
+	notification = response.ActivityCenterNotifications()[0]
+	s.Require().NotNil(notification)
+	s.Require().Equal(notification.Type, ActivityCenterNotificationTypeCommunityMembershipRequest)
+	s.Require().Equal(notification.MembershipStatus, ActivityCenterMembershipStatusAccepted)
+	s.Require().Equal(notification.Read, true)
+	s.Require().Equal(notification.Accepted, true)
+	s.Require().Equal(notification.Dismissed, false)
+
+	// Pull message and make sure org is received
+	err = tt.RetryWithBackOff(func() error {
+		response, err = s.alice.RetrieveAll()
+		if err != nil {
+			return err
+		}
+		if len(response.Communities()) == 0 {
+			return errors.New("community not received")
+		}
+		return nil
+	})
+
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+
+	s.Require().Len(response.RequestsToJoinCommunity, 1)
+	s.Require().Equal(communities.RequestToJoinStateAccepted, response.RequestsToJoinCommunity[0].State)
+
+	s.Require().Len(response.ActivityCenterNotifications(), 1)
+
+	notification = response.ActivityCenterNotifications()[0]
+	s.Require().NotNil(notification)
+	s.Require().Equal(notification.Type, ActivityCenterNotificationTypeCommunityRequest)
+	s.Require().Equal(notification.MembershipStatus, ActivityCenterMembershipStatusAccepted)
+	s.Require().Equal(notification.Read, false)
+	s.Require().Equal(notification.Accepted, false)
+	s.Require().Equal(notification.Dismissed, false)
+
+	s.Require().Len(response.Communities(), 1)
+	aliceCommunity := response.Communities()[0]
+
+	s.Require().Equal(community.ID(), aliceCommunity.ID())
+	s.Require().True(aliceCommunity.HasMember(&s.alice.identity.PublicKey))
+
+	// Community should be joined at this point
+	s.Require().True(aliceCommunity.Joined())
+
+	// Make sure the requests are not pending on either sides
+	requestsToJoin, err = s.bob.PendingRequestsToJoinForCommunity(community.ID())
+	s.Require().NoError(err)
+	s.Require().Len(requestsToJoin, 0)
+
+	requestsToJoin, err = s.alice.MyPendingRequestsToJoin()
+	s.Require().NoError(err)
+	s.Require().Len(requestsToJoin, 0)
```

Comment:
I think this long part about joining the community doesn't make sense in this test.
IMO the test should be simple and only test the sharing functionality:
1. owner: create community, no matter of Membershiprules)
2. owner: share community to Alice
3. alice: check the received message

It doesn't even matter if Alice joined the community. If the URL was received, then we're good.

wdyt?

Reply:
The long test was to test if a non admin shared community.
Because of the admin was the only member who could share a community.
But I guess since this is a completely new implementation I can do your suggestion.

Reply:
So I still think that it makes no sense to test all of this here, it's overkill.

> But I guess since this is a completely new implementation

Yep, sounds very reasonable.